### PR TITLE
Also import and export user_grid_outer_spacing in XML format and do not emit default values

### PR DESCRIPTION
**Pull request by tobiolo**

### CODE REVIEW

File: src/grid.h [hunk 1]
```diff
@@ -622,8 +622,17 @@ struct Grid {
...
         const int grid_border_width =
             cell == doc->rootgrid ? root_grid_spacing : user_grid_outer_spacing - 1;
-        Formatter(r, format, indent,
-                  wxString::Format(L"<grid bordercolor=\"0x%06X\">\n", bordercolor),
+
+        wxString xmlstr(L"<grid");
+        if (bordercolor != 0xA0A0A0) {
+            xmlstr.Append(wxString::Format(wxT(" bordercolor=\"0x%06X\""), bordercolor));
+        }
+        if (user_grid_outer_spacing != 3) {
```
Comment (aardappel) [inline src/grid.h:630]: `3` is probably a constant somewhere, and if not it should be?

File: src/system.h [hunk 2]
```diff
@@ -568,6 +570,7 @@ struct System {
                 c->AddGrid(1, numrows);
                 c->grid->bordercolor = std::stoi(
                     n->GetAttribute(L"bordercolor", L"0xA0A0A0").ToStdString(), nullptr, 0);
+                c->grid->user_grid_outer_spacing = wxAtoi(n->GetAttribute(L"outerspacing", L"3"));
```
Comment (aardappel) [inline src/system.h:573]: and here

File: src/grid.h (hunk 1 above)
Comment (tobiolo) [reply src/grid.h]: Definitely it should be! Thanks for your feedback. Pushed an extra commit for assigning the magic number to a global variable.

Comment (tobiolo) [reply src/grid.h]: Thanks for your review and time!

**Pull request merged**